assets/js/components/PostSearcherAutoSuggest.js

```diff
@@ -77,17 +85,41 @@ export default function PostSearcherAutoSuggest( {
 
 	useEffect( () => {
 		if ( debouncedValue !== '' ) {
+			setIsLoading?.( true );
 			API.get(
 				'core',
 				'search',
 				'post-search',
 				{ query: encodeURIComponent( debouncedValue ) },
 				{ useCache: false }
 			)
-				.then( setResults )
-				.catch( () => setResults( [] ) );
+				.then( ( res ) => setResults( res ) )
+				.catch( () => setResults( [] ) )
+				.finally( () => setIsLoading?.( false ) );
 		}
-	}, [ debouncedValue, setResults ] );
+	}, [ debouncedValue, setIsLoading ] );
+
+	useEffect( () => {
+		if ( ! searchTerm ) {
+			setResults( [] );
+		}
+	}, [ searchTerm ] );
+
+	const onKeyDown = useCallback(
+		( e ) => {
+			if ( ! unifiedDashboardEnabled ) {
+				return;
+			}
+			if ( e.keyCode === ESCAPE ) {
+				return onClose();
+			}
+
+			if ( e.keyCode === ENTER ) {
+				return onSelectCallback( searchTerm );
```

Comment:
@kuasha420 we also need to update the `onSelectCallback` callback to make it case-insensative:

```js
const foundMatch = results.find(
	( post ) =>
		post.post_title.toLowerCase() === value.toLowerCase()
);
```